release/models/hashing/openconfig-hashing.yang

```diff
@@ -0,0 +1,238 @@
+                    to calculate the hash.";
+            }
+        }
+        description "The IPv6 fields that should be used to
+            compute the hash.";
+    }
+
+    typedef hash-field-type-transport {
+        type enumeration {
+            enum NONE {
+                description "Do not use any transport fields in the
+                    calculation of the hash.";
+            }
+            enum SRC_PORT {
+                description "Use only the source port from the
+                    transport header in the calculation of the hash.";
+            }
+            enum DST_PORT {
+                description "Use only the destination port from the
+                    transport header in the calculation of the hash.";
+            }
+            enum BOTH {
+                description "Use both the source and
+                    destination ports from the transport header in
+                    the calculation of the hash.";
+            }
+            enum TARGET_DEFINED {
+                description "Use the default fields that are used by
+                    the device for calculating the hash.";
+            }
+        }
+        description "The trasport layer fields that should be used to
+            compute the hash.";
+    }
+
+    grouping hashing-inputs {
+        description
+            "Top level container for inputs to be used for hashing policies.";
+
+        leaf ipv4 {
+            type hash-field-type-ipv4;
+            description "The IPv4 fields that should be used to
+                compute the hash.";
+        }
+        leaf ipv6 {
+            type hash-field-type-ipv6;
+            description "The IPv6 fields that should be used to
+                compute the hash.";
+        }
+        leaf transport-ports {
+            type hash-field-type-transport;
+            description "The transport layer fields that should be used to
+                compute the hash.";
+        }
+    }
+
+    grouping hashing-policy-config {
+        description
+          "Configuration data for hashing policies.";
+
+        leaf name {
+            type string;
+            description
+              "The name of the hashing policy.
+
+              When a configured user-controlled policy is created by the
+              system, it is instantiated with the same name in the
+              /policies/policy[name]/state list.";
+        }
+
+        leaf algorithm {
+            type string;
+            description
+              "The algorithm used when calculating a hash based on packet headers";
+        }
+
+        leaf offset {
+            type uint64;
+            description
+                "The number of bits to rotate the output of the hash algorithm by
+                in order to get the final computed hash value";
+        }
+
+        leaf seed {
+            type uint64;
+            description
+              "The seed used to initialize the hash algorithm";
+        }
+
```

Comment:
```suggestion
        leaf symmetric {
            type boolean;
            default false;
            description
              "Hash is sematrical - returns same value regardless of traffic direction of same flow/session. Also regardless of ingress interface";
        }
        leaf persistent {
            type boolean;
            default false;
            description
              "In case of output inteface/forwarding-nh failure/removal, only flows that have used this interface/nexthop are re-hashed";
        }
```

Reply:
References to implementation needed for this?